default/AI/ShipDesignAI.py

```diff
@@ -1442,6 +1479,51 @@ def _partclass_in_design(self, partclass):
         """
         return any(part.partClass in partclass for part in self.parts)
 
+    def _calculate_weapon_strength(self, weapon):
+        # base damage
+        weapon_name = weapon.name
+        damage = weapon.capacity
+        # species modifier
+        weapons_grade, _, _ = CombatRatingsAI.get_piloting_grades(self.species)
+        species_modifier = AIDependencies.PILOT_DAMAGE_MODIFIER_DICT.get(weapons_grade, {}).get(weapon_name, 0)
+        # tech upgrades
+        try:
+            upgrades = AIDependencies.WEAPON_UPGRADE_DICT[weapon_name]
+        except KeyError:
+            global _raised_warnings
+            if weapon_name not in _raised_warnings:
+                _raised_warnings.add(weapon_name)
+                print_error(("WARNING: Encountered unknown weapon (%s): "
+                             "The AI will play on but damage estimates may be incorrect leading to worse performance. "
```

Comment:
For me in code context performance is performance of the code, I expect you mean some thing
`leading to wrong estimations of weapon streigths.`

trailing space in message should be removed